apiserver/registration.go

```diff
@@ -51,7 +51,7 @@ func (h *registerUserHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 
 	// TODO (stickupkid): Remove this nonsense, we should be able to get the
 	// service factory from the handler.
-	serviceFactory := h.ctxt.srv.shared.serviceFactoryGetter.FactoryForModel(st.ModelUUID())
+	serviceFactory := h.ctxt.srv.shared.serviceFactoryGetter.FactoryForModel(h.ctxt.srv.shared.controllerModelID)
```

Comment:
You should make a comment here, that the service factory is only using the services from the controller model, so any model would suffice here. I'd probably be more specific and say that we __should__ use the `controllerNS` here.

Reply:
not sure, the `FactoryForModel` specifically asks for the model UUID, so I don't think we should be passing in any old junk string just because we can.

Reply:
You could also set the type of the var to ControllerFactory so that we restrict the usage.

Reply:
@tlm good idea